File: hw/vizcaino/u3/hw28Strategy/hw28StrategyJava/Strategy/src/main/java/ec/edu/espe/strategy/SortingContext.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategy;

/**
 *
 * @author Adrian Vizcaino <The-Softwarriors at ESPE>
 */
public class SortingContext {
    private SortingStrategy ss;

    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }

    public SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n < 30) ss = new BubbleSort();
        if (n >= 30 && n < 100) ss = new InsertionSort();
        if (n >= 100) ss = new QuickSort();
        return ss;
    }
}

File: hw/palomo/u3/hw28Strategy/Strategy.js/controller/SortingContext.js
const BubbleSort = require('./BubbleSort');
const InsertionSort = require('./InsertionSort');
const QuickSort = require('./QuickSort');

class SortingContext {

    constructor() {
        this.sortingStrategy = null;
    }

    sort(data) {

        this.sortingStrategy = this.setSortStrategy(data.length);

        return this.sortingStrategy.sort(data);

    }

    setSortStrategy(size) {

        if (size > 0 && size < 30) {

            this.sortingStrategy = new BubbleSort();

        } else if (size >= 30 && size < 100) {

            this.sortingStrategy = new InsertionSort();

        } else {

            this.sortingStrategy = new QuickSort();

        }

        return this.sortingStrategy;

    }

}

module.exports = SortingContext;